code/modules/mob/living/carbon/human/appearance.dm

```diff
@@ -192,9 +219,23 @@
 			continue
 		if(gender == FEMALE && S.gender == MALE)
 			continue
-		if(!(species.name in S.species_allowed))
-			continue
-
-		valid_facial_hairstyles += facialhairstyle
+		if(species.flags & ALL_RPARTS) //If the user is a species who can have a robotic head...
+			if((species.name in S.species_allowed)) //If this is a facial hair style native to the user's species...
+				if(!user.client.prefs.rlimb_data["head"]) //Check to see if they have the default head.
+					valid_facial_hairstyles += facialhairstyle //Give them their facial hair styles if they do.
+					continue
+				else //If they don't have the default head, they shouldn't be getting any facial hair styles they wouldn't normally.
+					continue
+			else
+				if(!user.client.prefs.rlimb_data["head"]) //If the facial hair style is not native to the user's species, and they're using the default head, don't let them access it.
+					continue
+				else
+					if(("Human" in S.species_allowed)) //If the user has a robotic head and the facial hair style can fit humans, let them use it as a postiche for their humanoid robot head.
+						valid_facial_hairstyles += facialhairstyle
+					continue
+		else //If the user is not a species who can have robotic heads, use the default handling.
+			if(!(species.name in S.species_allowed))
```

Comment:
Thanks, I figured from what you said before I'd be better off taking the 'rather safer than sorry(er)' approach, but I'll strip the extra brackets.